pure_protobuf/oneof.py

```diff
@@ -0,0 +1,91 @@
+import dataclasses
+import functools
+from enum import Enum
+from typing import Any, Callable, Optional, Tuple
+
+
+@dataclasses.dataclass(frozen=True)
+class OneOfPartInfo:
+    name: str
+    type_: type
+    number: int
+
+
+class _OneOfAttrs(Enum):
+    SET_VALUE = "__set_value__"
+    PARTS = "__parts__"
+    FIELDS = "__fields__"
+
+
+def scheme(obj: 'OneOf_') -> Tuple[OneOfPartInfo, ...]:
+    return obj.__parts__
+
+
+def internals(self) -> Tuple[Any, Any, Any]:
```

Comment:
Wouldn't it be more readable to just use the attributes directly instead of wrapping it into a function returning `Any`?

Reply:
It's used only in this file for `__eq__`, `__hash__`, `__repr__` implementation and likely should be named `_internals`. Not supposed to be used somewhere else.